src/aclk/mqtt_websockets/mqtt_wss_client.c

```diff
@@ -300,56 +266,52 @@ static int cert_verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
 #define HTTP_HDR_TERMINATOR "\x0D\x0A\x0D\x0A"
 #define HTTP_CODE_LEN 4
 #define HTTP_REASON_MAX_LEN 512
-static int http_parse_reply(mqtt_wss_client client, rbuf_t buf)
+static int http_parse_reply(rbuf_t buf)
 {
-    char *ptr;
     char http_code_s[4];
-    int http_code;
     int idx;
 
     if (rbuf_memcmp_n(buf, PROXY_HTTP, strlen(PROXY_HTTP))) {
         if (rbuf_memcmp_n(buf, PROXY_HTTP10, strlen(PROXY_HTTP10))) {
-            mws_error(client->log, "http_proxy expected reply with \"" PROXY_HTTP "\" or \"" PROXY_HTTP10  "\"");
+            nd_log(NDLS_DAEMON, NDLP_ERR, "http_proxy expected reply with \"" PROXY_HTTP "\" or \"" PROXY_HTTP10  "\"");
             return 1;
         }
     }
 
     rbuf_bump_tail(buf, strlen(PROXY_HTTP));
 
     if (!rbuf_pop(buf, http_code_s, 1) || http_code_s[0] != 0x20) {
-        mws_error(client->log, "http_proxy missing space after \"" PROXY_HTTP "\" or \"" PROXY_HTTP10  "\"");
+        nd_log(NDLS_DAEMON, NDLP_ERR, "http_proxy missing space after \"" PROXY_HTTP "\" or \"" PROXY_HTTP10  "\"");
         return 2;
     }
 
     if (!rbuf_pop(buf, http_code_s, HTTP_CODE_LEN)) {
-        mws_error(client->log, "http_proxy missing HTTP code");
+        nd_log(NDLS_DAEMON, NDLP_ERR, "http_proxy missing HTTP code");
         return 3;
     }
 
     for (int i = 0; i < HTTP_CODE_LEN - 1; i++)
         if (http_code_s[i] > 0x39 || http_code_s[i] < 0x30) {
-            mws_error(client->log, "http_proxy HTTP code non numeric");
+            nd_log(NDLS_DAEMON, NDLP_ERR, "http_proxy HTTP code non numeric");
             return 4;
         }
 
     http_code_s[HTTP_CODE_LEN - 1] = 0;
-    http_code = atoi(http_code_s);
+    int http_code = atoi(http_code_s);
```

Comment:
Now that we are doing cleanup and adjusting the code, we could also use `str2i` instead `atoi`.